### DISCUSSION TITLE
Multi-line Comment Parsing

### DISCUSSION THREAD

**Discussion by tinygiant98** [tree-sitter/tree-sitter#1173] [Q&A] [answered]
I found yet another oddity that my nearly non-existent regex skills have yet to be able to overcome.  Since I'm using tree-sitter's CST as a source for syntax highlighting, it has to work as you type, which it does for the most part.  I've found an issue with multi-line comments, however. Here's the entry that identifies multi-line coments from `grammar.js`:
```js
    comment: $ => token(choice(
      seq('//', /(\\(.|\r?\n)|[^\\\n])*/),
      seq(
        '/*',
        /[^*]*\*+([^/*][^*]*\*+)*/,
        '/'
      )
    )),
```
This correctly identifies fully enclosed multi-line comments.  However, if the user is typing and enters the multi-line comment opener `/*`, the text does not parse as a comment until the user types in the multi-line comment closer `*/`.  Is there a way to identify all characters to the end of the file (assuming `*/` doesn't exist in the file) as a comment until the user types in the closer?  I've done a bit of research, but it seems that js doesn't accept the primary methods for doing this, including `\z`.  I found two ways that worked in regex sandboxes, but neither worked in the grammar.

**Comment by tek** [comment]
you can make the closing delimiter `optional`

**Reply by tinygiant98** [reply]
Thanks for the reply.  I tried that yesterday to no avail as the current syntax requires matching at least one asterisk.  If there's not yet an asterisk in the text, it will never match.  When there is an asterisk, it'll stop there and capture to that point, even if it's not a closer.  In Regex sandboxes, the very simple `/(.*)/s` worked wonderfully, but failed to work in the grammar (stops at `\n`, so apparently `/s` doesn't register correctly, even though there's no error in the generate).

**Comment by tinygiant98** [comment]
So, this works, but it's hacky and not a true "match anything" resolution, which I'll keep looking for.  I tried `/([^]+)/`, which should work but threw an unclosed character class error in the generate.  Adding in an obscure character not likely to be encounters (`°` in this case, but could be anything in the high end of ascii), gets it to work during live typing.  -> `/([^°]+/`

**Reply by tek** [reply]
why don't you include `\n` explicitly?

**Reply by tinygiant98** [reply]
Tried that, couldn't get it to work.  I also suck at regex, so there's that.

**Comment by ahlinc** [comment, marked as answer]
A bit of more info for everyone who will see this question:
* https://github.com/tree-sitter/tree-sitter/discussions/2094#discussioncomment-5082953
* https://github.com/tree-sitter/tree-sitter/discussions/1252#discussioncomment-1222142
